tests/migrations/test_operations.py

```diff
@@ -1813,6 +1826,7 @@ def test_remove_constraint(self):
         operation.state_forwards("test_removeconstraint", new_state)
         self.assertEqual(len(new_state.models["test_removeconstraint", "pony"].options['constraints']), 0)
         Pony = new_state.apps.get_model("test_removeconstraint", "Pony")
+        self.assertEqual(len(Pony._meta.constraints), 0)
```

Comment:
@carltongibson doesn't this assertion fails for your without the `state.reload_model` addition in `RemoveConstraint.state_forwards`?

`RemoveConstraint` happened to suffer from an analogous issue; if you have a model with multiple constraints and remove both the first removed constraint would resurface. I guess I could add extra assertions for that as well if you'd like.

Reply:
Hi @charettes. 

(Grrr. GitHub isn't working today...) 

Short answer, No. 

Resetting to 9374b3820ba98a8a6809bf82e49e7460920809a9 and just applying the changes to `test_operations.py` I get a failure in `test_add_constraint()` but not `test_remove_constraint()`.

> RemoveConstraint happened to suffer from an analogous issue...

It looks as if the test model only has the single constraint (from `set_up_test_model()`) so it wouldn't hit the error here. 

Hadn't got as far as an actual opinion on whether we need the extra coverage yet... 🙂

Reply:
Ahh got it. The assertion happens to pass because the pony model is only materialized from the project state once after the operation is applied forward. I'll adjust the `test_remove_constraint` test to be more thorough by creating and removing multiple constraints.